### DISCUSSION TITLE
DB options

### DISCUSSION THREAD

**Discussion by ihoro-epam** [abdukhashimov/student_aggregator#13] [Ideas]
Assuming that we will try to implement layered arch DB selection is not so important. But it would be great to choose one db for development.

Options we can consider are:
1. Postgresql
2. Mongodb
3. Redis
4. SQLite

@davit312 @maxwww @ADrema @abdukhashimov @arman42117 Please provide your thoughts on this problem.

**Comment by ihoro-epam** [comment]
As long as we do not have final data structure SQL dbs will add a bit of complexity with all this migrations etc. In opposite schemaless dbs will allow as to reduce this complexity and concentrate our effort on business problem.

My personal selection is MongoDB. NoSQL document db with no data schema. Easy to run and scale. A lot of tools to integrate.

**Reply by abdukhashimov** [reply]
I also support mongodb, I think we can implement PoC with mongodb.

**Reply by abdukhashimov** [reply]
The way it should be implemented, is the best possible way that it just works, no structure and even clean code is needed.
Why? just to proof the concept, then we can clean up the implementation

**Reply by maxwww** [reply]
It's OK for me too
But if we have some extra time after implementing the project it will be nice to figure out how to handle migrations in SQL dbs and how to organize db repository and with ORM or without.

**Reply by abdukhashimov** [reply]
Yeah, let's give it a try with schemaless architecture, at the end we will have a rough schema generated that works pretty much for all cases. And adding new types of db should not be a problem as long as we have the interfaces. Regarding ORM, I prefer not using ORM 😄 . It is much better with builders or sql compilers INMHO
